src/trace/implementations/chunker.rs

```diff
@@ -0,0 +1,366 @@
+//! Organize streams of data into sorted chunks.
+
+use std::collections::VecDeque;
+use timely::communication::message::RefOrMut;
+use timely::Container;
+use timely::container::columnation::{Columnation, TimelyStack};
+use timely::container::{ContainerBuilder, PushInto, SizableContainer};
+use crate::consolidation::{consolidate_updates, ConsolidateContainer};
+use crate::difference::Semigroup;
+
+/// Chunk a stream of vectors into chains of vectors.
```

Comment:
Is this materially different from a capacity container builder? If they are very similar, having one fewer abstraction is great! If there is a really important distinction, we should write it down so that I don't accidentally optimize this away in the future. :D

Reply:
Edit: perhaps rather compared to *consolidating* container builder, as it doesn't just choose a capacity.

Reply:
Not very different, but somewhat incompatible! The consolidating container builder has a `PushInto` implementation for individual items, but here we want to push whole containers (`RefOrMut<...>`). Since the other implementation already exists, we cannot define a impl for just `RefOrMut` because that would conflict.

There is a path out of this mess, but I don't know if we want to go down that route: We could introduce a `ContainerMarker<C>(C)` struct that we can use to specialize the `PushInto` implementation for the consolidating container builder. Since nothing that's further up in the hierarchy can know this type, Rust knows that there can't be conflicting implementations. If we had that, we might be able to remove the `VecChunker`.

That aside, having $n$ different implementations that are roughly the same isn't great, and I believe with a bit more effort we could tidy it up.